node/router/src/inbound.rs

```diff
@@ -349,20 +349,17 @@ pub trait Inbound<N: Network>: Reading + Outbound<N> {
     }
 
     /// Handles a `PeerResponse` message.
-    fn peer_response(&self, _peer_ip: SocketAddr, peers: &[SocketAddr]) -> bool {
-        // Check if the number of peers received is less than MAX_PEERS_TO_SEND.
-        if peers.len() > MAX_PEERS_TO_SEND {
+    fn peer_response(&self, _peer_ip: SocketAddr, peers: Vec<SocketAddr>) -> bool {
+        // Check if the number of peers received is less than MAX_PEERS_TO_SEND or empty.
+        if peers.len() > MAX_PEERS_TO_SEND || peers.is_empty() {
```

Comment:
Could an empty peer list be valid behaviour? I would remove the change as it does not feel like durable/important.

Reply:
Ah, fair, I didn't realize it constituted a protocol violation.